sdks/java/core/src/main/java/org/apache/beam/sdk/util/Secret.java

```diff
@@ -33,4 +38,48 @@ public interface Secret extends Serializable {
    * @return The secret as a byte array.
    */
   byte[] getSecretBytes();
+
+  static Secret parseSecretOption(String secretOption) {
+    Map<String, String> paramMap = new HashMap<>();
+    for (String param : secretOption.split(";", -1)) {
+      String[] parts = param.split(":", 2);
+      if (parts.length == 2) {
+        paramMap.put(parts[0], parts[1]);
+      }
+    }
+
+    if (!paramMap.containsKey("type")) {
+      throw new RuntimeException("Secret string must contain a valid type parameter");
+    }
+
+    String secretType = paramMap.get("type");
+    paramMap.remove("type");
+
+    if (secretType == null) {
```

Comment:
Do you mean `PreConditions.checkNotNull(secretType, "Secret string must contain a valid value for type parameter");`

Reply:
If I don't have the explicit null check, this will not compile unfortunately:

```
/Users/dannymccormick/beam/sdks/java/core/src/main/java/org/apache/beam/sdk/util/Secret.java:61: error: [dereference.of.nullable] dereference of possibly-null reference secretType
    switch (secretType.toLowerCase()) {
```

It seems cleaner to just leave it.